### PR TITLE
Support cjs/mjs/cts/mts glue extensions

### LINKED ISSUE #78: Handling *.mts files
### 🤔 What's the problem you're trying to solve?

Currently *.mts files (and probably *.mjs, *.cjs and *.cts) are ignored as glue files:
```
[Info  - 5:31:37 PM] * Found 2 feature file(s) in ["features/**/*.feature"]
[Info  - 5:31:37 PM] * Found 50 steps in those feature files
[Info  - 5:31:37 PM] * Found 0 glue file(s) in ["features/step-definitions/**/*.mts"]
[Info  - 5:31:37 PM] * Found 0 parameter types in those glue files
[Info  - 5:31:37 PM] * Found 0 step definitions in those glue files
```

### ✨ What's your proposed solution?

To add those files extensions to the supported ones, I think it should probably be added in https://github.com/cucumber/language-server/blob/main/src/fs.ts and  https://github.com/cucumber/language-server/blob/main/src/CucumberLanguageServer.ts to support those file extensions.

### ⛏ Have you considered any alternatives or workarounds?

Yes, to use simple ts files but I would prefer to keep .mts files

### 📚 Any additional context?
\-

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/ISSUE_TEMPLATE/feature_request.md)*

### PR DESCRIPTION
**Pull request by surc54**
### 🤔 What's changed?

This PR adds support for glue files with the following file extensions:
- Javascript: `*.cjs`, `*.mjs`
- Typescript: `*.cts`, `*.mts`

### ⚡️ What's your motivation? 

I wanted to use `*.mjs` glue files to keep consistency with the rest of my repository.

Fixes #78 

### 🏷️ What kind of change is this?

- :zap: New feature (non-breaking change which adds new behaviour)

### ♻️ Anything particular you want feedback on?

I decided not to edit the default settings since I'm not aware of what it might affect.

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [x] I've changed the behaviour of the code
  - [x] I have added/updated tests to cover my changes.
- [x] ~My change requires a change to the documentation.~ N/A
  - [x] ~I have updated the documentation accordingly.~ N/A
- [x] Users should know about my change
  - [x] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

### CODE REVIEW

**Review by mpkorstanje — APPROVED** [review]

**Comment by mpkorstanje** [comment]
LGTM. Though I'm not an expert.

@xeger can you release this along with VS code?

**Comment by mpkorstanje** [comment]
@surc54 looks like CI disagrees.

**Comment by surc54** [comment]
@mpkorstanje Hmmm, those crashes are quite interesting...

I believe I tracked it down to this line in `vscode-languageserver`: https://github.com/microsoft/vscode-languageserver-node/blob/07b9d293ea775dfd263562da3e8321d40397a3d0/server/src/node/main.ts#L89

That piece of code seems to be responsible for watching the parent process, and in the event that the parent process seems to have been killed, it will kill the current process as well.

If this is the culprit, that means this PR just exposed the problem by pushing the test running time past the 3s barrier.

I've tried to mitigate this by changing the processId parameter to a string (58d0705f25400e474883b8085a2c49ced4ed3882) -- but please note that I'm not aware of the consequences of this change (this is my first dabble into vscode extensions)

File: test/CucumberLanguageServer.test.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ describe('CucumberLanguageServer', () => {
 
     const initializeParams: InitializeParams = {
       rootUri: `file://${process.cwd()}`,
-      processId: 1,
+      processId: 'N/A' as unknown as number, // This id is used by vscode-languageserver. Forcing this to be a string so that the watchdog responsible for watching this process does not run.
```
Comment (mpkorstanje) [inline test/CucumberLanguageServer.test.ts:53]: Instead of using a string, would `NaN` work as well?

Comment (surc54) [reply test/CucumberLanguageServer.test.ts]: Oh, indeed it does -- good idea. Updated here 658f17727e32807f2a82dce89eed6bfb614315d0

**Comment by mpkorstanje** [comment]
Thanks for the explanation! I don't know much about VSCode either. This project is pretty much all made up by drive-by-volunteers. Looking at the code though it seems this is the intended solution.

**Comment by aslakhellesoy** [comment]
Hi @surc54,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**